### PR TITLE
saleae-logic-2: init at 2.3.33

### PR DESCRIPTION
**Pull request by j-hui**
###### Motivation for this change

Saleae has released version 2 of their logic analyzer software, Logic 2, though the original version Logic 1 still appears to be available.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/saleae-logic-2/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib, fetchurl, appimageTools }:
+let name = "saleae-logic-2";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/saleae-logic-2/default.nix:2]:
```suggestion
let
  name = "saleae-logic-2";
```
Please format the file with nixpkgs-fmt.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -4755,6 +4755,12 @@
     githubId = 40566146;
     name = "Jonas Braun";
   };
+  j-hui = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4758]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

Comment (j-hui) [reply maintainers/maintainer-list.nix:4758]: Thanks for the review! Should I force push the new commits?

Comment (SuperSandro2000) [reply maintainers/maintainer-list.nix:4758]: yes, please

Comment (j-hui) [reply maintainers/maintainer-list.nix:4758]: Done! I was unable to sort it alphabetically globally, because the maintainer list seems to already be out of order: `j-brn`, `j-keck`, and `j03` are positioned between `jasoncarr` and `jackgerritts`, for some reason.
